coding-standards-and-naming.rst

```diff
@@ -188,6 +226,192 @@ happen with the agreement of the parties involved). This document does not try
+   even then, try avoiding it.
+
+#. You SHOULD use ``PHP_*`` macros in the PHP source, and ``ZEND_*`` macros in
+   the Zend part of the source. Although the ``PHP_*`` macros are mostly aliased
+   to the ``ZEND_*`` macros it gives a better understanding on what kind of
+   macro you're calling.
+
+#. You MUST NOT define functions that are not available. For instance, if a
+   library is missing a function, do not define the PHP version of the function,
+   and do not raise a run-time error about the function not existing. End users
+   should use ``function_exists()`` to test for the existence of a function.
+
+#. You SHOULD use ``emalloc()``, ``efree()``, ``estrdup()``, instead of their
+   standard C library counterparts. These functions implement an internal
+   "safety-net" mechanism that ensures the deallocation of any unfreed memory at
+   the end of a request. They also provide useful allocation and overflow
+   information while running in debug mode.
+
+   In almost all cases, memory returned to the engine must be allocated using
+   ``emalloc()``.
+
+   The use of ``malloc()`` SHOULD be limited to cases where a third-party
+   library may need to control or free the memory, or when the memory in
+   question needs to survive between multiple requests.
+
+#. The return type of "is" or "has" style functions SHOULD be ``bool`` which
+   return a "yes"/"no" answer. `zend_result` is an appropriate return value for
+   functions that perform some operation that may succeed or fail.
+
+Variable Names
+==============
+
+Variable names MUST be meaningful. One letter variable names SHOULD be avoided,
+except for places where the variable has no real meaning or a trivial meaning
+(e.g. ``for (i=0; i<100; i++) ...``).
+
+Variable names MUST be in lowercase. Use underscores to separate between words.
+
+Internal Function Naming Conventions
+====================================
+
+The main module source file MUST be named ``modulename.c``.
+
+Header files that are used by other sources must be named ``php_modulename.h``.
+
+Functions that are part of the external API MUST be named
+``php_modulename_function()`` to avoid symbol collision. They MUST be in
+lowercase, with words underscore delimited. Exposed API MUST be defined in
+``php_modulename.h``:
+
+.. code::
+
+   PHPAPI char *php_session_create_id(PS_CREATE_SID_ARGS);
+
+Unexposed module function MUST be ``static`` and MUST NOT be defined in
+`php_modulename.h`:
+
+.. code::
+
+   static int php_session_destroy()
+
+Syntax and indentation
+======================
+
+You SHOULD Use K&R-style. When you write code that goes into the core of PHP or
+one of its standard modules, please maintain the K&R style.
+
+Be generous with whitespace and braces. Keep one empty line between the variable
+declaration section and the statements in a block, as well as between logical
+statement groups in a block. Maintain at least one empty line between two
+functions, preferably two. Always prefer:
```

Comment:
> preferably two

We don't do this anywhere. Can we drop that?

Reply:
I'm okay dropping that, but this PR isn't about changing existing rules. Perhaps it makes sense to put these changes in the RFC that I will send to internals?